common/stack_tracer.go

```diff
@@ -60,6 +61,83 @@ func TraceAsmCaller(
 	return src, nil
 }
 
+// TraceAllAsmCaller correctly tracks all possible function calls in the execution stack.
+func TraceAllAsmCaller(
+	filePath string,
+	graph asmparser.CallGraph,
+	function string,
+	endCond func(string) bool,
+) ([]*analyzer.CallStack, error) {
+	var segment asmparser.Segment
+	for _, seg := range graph.Segments() {
+		if seg.Label() == function {
+			segment = seg
+			break
+		}
+	}
+	if segment == nil {
+		return nil, fmt.Errorf("could not find %s in %s", function, filePath)
+	}
+	sources := make([]*lifo.Stack[asmparser.Segment], 0)
+	currentStack := lifo.Stack[asmparser.Segment]{}
+	seen := make(map[asmparser.Segment]bool)
+
+	var visit func(segment asmparser.Segment)
+
+	visit = func(segment asmparser.Segment) {
+		if seen[segment] {
+			return
+		}
+		seen[segment] = true
+		currentStack.Push(segment)
+
+		if len(sources) >= 10 {
+			return
```

Comment:
```suggestion
                         currentStack.Pop()
			seen[segment] = false
			return
```

Reply:
I have updated the code.
Also added single path for runtime and internal package.